allennlp/data/fields/knowledge_graph_field.py

```diff
@@ -0,0 +1,370 @@
+                'contains_exact_token_match',
+                'lemma_match',
+                'contains_lemma_match',
+                'edit_distance',
+                'related_column',
+                'related_column_lemma',
+                'span_overlap_fraction',
+                ]
+        self._feature_extractors: List[Callable[[str, List[Token], Token, int, List[Token]], float]] = []
+        for feature_extractor_name in feature_extractors:
+            extractor = getattr(self, '_' + feature_extractor_name, None)
+            if not extractor:
+                raise ConfigurationError(f"Invalid feature extractor name: {feature_extractor_name}")
+            self._feature_extractors.append(extractor)
+
+        if not linking_features:
+            # For quicker lookups in our feature functions, we'll additionally store some
+            # dictionaries that map entity strings to useful information about the entity.
+            self._entity_text_map: Dict[str, List[Token]] = {}
+            for entity, entity_text in zip(knowledge_graph.entities, self.entity_texts):
+                self._entity_text_map[entity] = entity_text
+
+            self._entity_text_exact_text: Dict[str, Set[str]] = {}
+            for entity, entity_text in zip(knowledge_graph.entities, self.entity_texts):
+                self._entity_text_exact_text[entity] = set(e.text for e in entity_text)
+
+            self._entity_text_lemmas: Dict[str, Set[str]] = {}
+            for entity, entity_text in zip(knowledge_graph.entities, self.entity_texts):
+                self._entity_text_lemmas[entity] = set(e.lemma_ for e in entity_text)
+            self.linking_features = self._compute_linking_features()
+        else:
+            self.linking_features = linking_features
+
+    @overrides
+    def count_vocab_items(self, counter: Dict[str, Dict[str, int]]):
+        if self._include_in_vocab:
+            for indexer in self._token_indexers.values():
+                for entity_text in self.entity_texts:
+                    for token in entity_text:
+                        indexer.count_vocab_items(token, counter)
+
+    @overrides
+    def index(self, vocab: Vocabulary):
+        self._indexed_entity_texts = {}
+        for indexer_name, indexer in self._token_indexers.items():
+            indexer_arrays = []
+            for entity_text in self.entity_texts:
+                indexer_arrays.append([indexer.token_to_indices(token, vocab) for token in entity_text])
+            self._indexed_entity_texts[indexer_name] = indexer_arrays
+
+    @overrides
+    def get_padding_lengths(self) -> Dict[str, int]:
+        num_entities = len(self.entity_texts)
+        num_entity_tokens = max(len(entity_text) for entity_text in self.entity_texts)
+
+        if self._max_table_tokens:
+            # This truncates the number of entity tokens used, enabling larger tables (which can
+            # have thousands of entities or more than 50 tokens each) to fit in memory,
```

Comment:
of more than 50 tokens each?

Reply:
No, the issue is that the table can be big either because there are a ton of entities or because there are a ton of tokens per entity (or both, but either one can be an issue).  I clarified the wording here.